### PR TITLE
Potential fix for #2238 Kid Icarus Uprising

### PR DESCRIPTION
**Pull request by SigmaVirus**
According to @mailwl ReadSavedData needs to be stubbed in order to pass the loop in Kid Icarus.

### CODE REVIEW

File: src/core/hle/service/cecd/cecd.cpp [hunk 1]
```diff
@@ -42,6 +42,13 @@ void GetChangeStateEventHandle(Service::Interface* self) {
 
     LOG_WARNING(Service_CECD, "(STUBBED) called");
 }
+    
+void OpenAndRead(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    cmd_buff[1] = -1;
+    LOG_WARNING(Service_CECD, "(STUBBED) called");
+}
```
Comment (jroweboy) [inline src/core/hle/service/cecd/cecd.cpp]: Trailing whitespace

**Comment by mailwl** [comment]
it's mostly the same as @wwylele hack with https://github.com/citra-emu/citra/blob/master/src/core/hle/service/nwm/nwm_uds.cpp#L91
not implemented yet, so return error. I dislike it too, but this will be fixed, when @wwylele implement cecd

**Comment by Subv** [comment]
That one at least has documentation and a sensible error code.

**Comment by mailwl** [comment]
true :)
CEC will not work, if WiFi is off, but "-1" error includes "Fatal" i think, it will be good to return proper code, not to break other games

**Comment by SigmaVirus** [comment]
OK I get it, it's a bad and awful cursed hack to get a game running.

But I just followed what @mailwl said in issue #2238 which he implemented in his own fork (https://github.com/mailwl/citra/tree/kid/src/core/hle/service/cecd) and thought it would solve the issue.

If you don't like this and feel it breaks more than it solves I could close the issue, but I just wanted to help out instead of just complaining about it.

**Comment by mailwl** [comment]
it's not bad itself i think, because Kid Icarus really cool to be emulated by Citra, but incompleted. i did just workaround, but Citra needs proper code

**Comment by wwylele** [comment]
Just want to correct what @mailwl said: CEC does work even if wifi is off. It allows game to check the mailbox offline.

**Comment by mailwl** [comment]
SigmaVirus, did you change error core to prorer one? looks like stop PR from merging

**Comment by SigmaVirus** [comment]
@mailwl 
While it's true that I'm an educated systems developer and database designer I got literally 0 experience in coding emulators, let alone C++

I really don't know how to change the error code to a proper one and I'm sorry but I didn't think this would be such a big problem when I made this PR.

I have been able to try these changes out practically so I know they work.
The game starts, can be played, saved etc. and I haven't seen any other games suffer from these changes (at least not from my collection)

It would be nice to solve this properly however, and I would in a heartbeat if I could.

Kid Icarus Uprising is a really cool game and it would be nice having all my games in the collection bootable and playable but since I don't have what it takes to make the changes I see three options here:
1. Close this PR
2. Merge it and see it as a temporary workaround that needs to be corrected
3. Someone else more experienced must help me and take over and finish it properly

**Comment by mailwl** [comment]
i have a loader to 3ds https://github.com/mailwl/IDA_plugin_3ds for IDA, but it's without signatures. it can do some help, write me on my mail mailwl@gmail.com and i send this strongly profibited file

**Comment by mailwl** [comment]
ah, the games itself dont have  proper error codes, it just does left SHL to 31 bits, to leave only -1

**Comment by SigmaVirus** [comment]
It's unlikely this is going somewhere right now so I'll just close this because I don't have what it takes to fix it.
Maybe someone with more expertise will pick this up and solve it.

**Pull request closed without merging**